### PR TITLE
loopout: fix negative reported fees

### PR DESCRIPTION
**Pull request by bhandras**
#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: sweepbatcher/sweep_batch.go [hunk 1]
```diff
@@ -1158,9 +1158,8 @@ func (b *batch) monitorConfirmations(ctx context.Context) error {
 func getFeePortionForSweep(spendTx *wire.MsgTx, numSweeps int,
 	totalSweptAmt btcutil.Amount) (btcutil.Amount, btcutil.Amount) {
 
-	totalFee := spendTx.TxOut[0].Value - int64(totalSweptAmt)
-	feePortionPerSweep := (int64(totalSweptAmt) -
-		spendTx.TxOut[0].Value) / int64(numSweeps)
+	totalFee := int64(totalSweptAmt) - spendTx.TxOut[0].Value
+	feePortionPerSweep := totalFee / int64(numSweeps)
```
Comment (starius) [inline sweepbatcher/sweep_batch.go:1162]: The function is pure. Let's add unit tests for it?

Comment (bhandras) [reply sweepbatcher/sweep_batch.go:1162]: Yeah, good point! Added.

**Review by hieblmi — APPROVED** [review]

File: loopout.go [hunk 2]
```diff
@@ -456,9 +456,11 @@ func (s *loopOutSwap) handlePaymentResult(result paymentResult) error {
 		return nil
 
 	case result.status.State == lnrpc.Payment_SUCCEEDED:
-		s.cost.Server += result.status.Value.ToSatoshis() -
-			s.AmountRequested
-		s.cost.Offchain += result.status.Fee.ToSatoshis()
+		if result.status.Preimage == s.SwapContract.Preimage {
```
Comment (starius) [inline loopout.go:459]: Please add a comment about this check (about the scenarios).

**Review by starius — APPROVED** [review]
LGTM 🏆

Added a comment

File: loopout.go (hunk 2 above)
Comment (bhandras) [reply loopout.go]: Instead I added a flag to indicate this is a swap payment and some comments explaining what we're doing when calculating the costs.

**Pull request merged**